src/common/path.hpp

```diff
@@ -43,7 +43,7 @@ enum class Perms {
 	Others_exec,
 };
 
-error::Error Permissions(const string &file_path, vector<Perms> perms);
+error::Error Permissions(const string &file_path, vector<Perms> perms, bool warn_on_change = false);
```

Comment:
Maybe the new argument should be an `enum class` with two members? It's a bit confusing to just see `true` when it's called, especially for this argument which is a bit unusual for this type of function.